lib/IDE/CompletionInstance.cpp

```diff
@@ -276,6 +277,39 @@ static bool areAnyDependentFilesInvalidated(
       });
 }
 
+/// Get interface hash of \p SF including the type members in the file.
+///
+/// This is needed because SouceFile::getIntefaceHash() doesn't digest type body
+/// fingerprints in the file.
+void getInterfaceHashIncludingTypeMembers(SourceFile *SF,
+                                          llvm::SmallString<32> &str) {
+  /// FIXME: Gross. Hashing multiple "hash" values.
+  llvm::MD5 hash;
+  SF->getInterfaceHash(str);
+  hash.update(str);
+
+  class FingerprintCollector: public ASTWalker {
+    llvm::MD5 &Hash;
+
+  public:
+    FingerprintCollector(llvm::MD5 &Hash): Hash(Hash) {}
+
+    bool walkToDeclPre(Decl *D) override {
+      if (auto *IDC = dyn_cast<IterableDeclContext>(D)) {
+        if (auto fp = IDC->getBodyFingerprint())
+          Hash.update(*fp);
+      }
+      return true;
+    }
+  } collector(hash);
+
+  SF->walk(collector);
```

Comment:
This will force delayed parsing of all type and extension bodies. Is that OK?

Reply:
Ah, you mean function bodies, right? It's not good.

Reply:
Updated to only walk into `IterableDeclContext`.